R/as-nlists.R

```diff
@@ -35,11 +35,10 @@ as_nlists.list <- function(x, ...) {
 #' @export
 as_nlists.mcmc <- function(x, ...) {
   chk_unused(...)
-  x <- complete_terms(x)
-  x <- lapply(1:nrow(x), function(iter, x) subset(x, iter), x = x)
-  x <- lapply(x, as_nlist)
-  class(x) <- "nlists"
-  x
+  if(!requireNamespace("mcmcr", quietly = TRUE)) {
+    err("Package 'mcmcr' must be installed.")
+  }
```

Comment:
```suggestion
  rlang::check_installed("mcmcr", "to convert mcmc objects to nlist.")
```

and perhaps move rlang to "imports"?